src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala

```diff
@@ -11,11 +11,55 @@ import org.apache.spark.sql.sources.InsertableRelation
 import org.apache.spark.sql.types.{DataTypes, StructType}
 import org.apache.spark.sql.{Row, SQLContext}
 import com.cognite.spark.datasource.SparkSchemaHelper._
+import io.circe.Decoder.Result
+import io.circe.{Decoder, Encoder, HCursor, Json}
 import org.apache.spark.datasource.MetricsSource
 
 import scala.concurrent.ExecutionContext
 
-case class PostAssetsDataItems[A](items: Seq[A])
+sealed case class Type(id: Long, name: String, fields: Seq[FieldTypes.Field])
+
+object FieldTypes {
+  case class Field(id: Long, name: String, valueType: String, value: String)
+  case class DoubleIsTooLargeForJSON(field: String)
+      extends Throwable(s"Double is too large for JSON in data type field $field.")
+
+  implicit val fieldDecoder: Decoder[Field] = new Decoder[Field] {
+    override def apply(c: HCursor): Result[Field] =
+      for {
+        id <- c.downField("id").as[Long]
+        name <- c.downField("name").as[String]
+        valueType <- c.downField("valueType").as[String]
+        value <- valueType match {
+          case "String" => c.downField("value").as[String]
+          case "Long" => c.downField("value").as[Long].map(_.toString)
+          case "Double" => c.downField("value").as[Double].map(_.toString)
+          case "Boolean" => c.downField("value").as[Boolean].map(_.toString)
+        }
+      } yield {
```

Comment:
I'm personally not a fan of curly brackets here, I prefer `yield Field(...)` and I think that's what we usually do, but we may not be consistent about this at the moment. I don't have any strong feelings about it.

Reply:
Agreed, fits better with the no-redundant-parenthesis rule from scalafmt.